complexify-android/src/main/java/co/infinum/complexify/Complexify.kt

```diff
@@ -0,0 +1,95 @@
+package co.infinum.complexify
+
+import co.infinum.complexify.Constants.CHARSETS_ARRAY
+import co.infinum.complexify.Constants.DEFAULT_BAN_LIST
+import kotlin.math.ln
+import kotlin.math.pow
+
+/**
+ * Class for calculation of password complexity.
+ *
+ * @param banMode             Use strict or loose comparisons for banned passwords.
+ * @param strengthScaleFactor Required password strength multiplier. Bigger factor = more complex password required.
+ * @param minimumChars        Minimum password length.
+ * @param banList             A list of banned passwords which will reduce input password's complexity to 0 if the password contains one of them. The ban list is case-insensitive. The default values are generated from 500 worst passwords and 370 Banned Twitter lists: https://wiki.skullsecurity.org/Passwords
+ */
+class Complexify @JvmOverloads constructor(
+    var banMode: ComplexifyBanMode = ComplexifyBanMode.STRICT,
+    var strengthScaleFactor: Double = 1.0,
+    var minimumChars: Int = 8,
+    var banList: Array<String> = DEFAULT_BAN_LIST
+) {
+
+    companion object {
+        private const val MIN_COMPLEXITY = 49 // 12 chars with Upper, Lower and Number
+        private const val MAX_COMPLEXITY = 120 // 25 chars, all charsets
+        private const val MAX_PERCENTAGE = 100.0
+    }
```

Comment:
This whole thing bothers me. It's very strict on publicly exposed API and allows no way of easy adjustment. 
In a situation where we add more params or remove some, each call site should be adjusted. 
Developers integrating it will be loud and furious in tears.
I suggest:

- create a Configuration interface
- create a ConfigurationFactory that must be able to create a DefaultConfiguration data class (that implements Configuration interface)
- make DefaultConfiguration a default parameter of Complexify constructor.
This allows shorter call sites with default parameters, enables developers to make custom Configurations when and if needed and ConfigurationFactory can be expanded easily to provide perhaps some commonly used Configurations.

???
Profit.